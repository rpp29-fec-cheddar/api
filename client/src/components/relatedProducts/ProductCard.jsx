import React from 'react';
import EachCard from './EachCard.jsx';

const ProductCard = (props) => {

  let combine = props.detailInfo.reduce((map, value) => {
    map[value.product_id] = value;
    return map;
  }, {});
  let relevantProps = [];
  for (let i = 0; i < props.info.length; i++) {
    let items = combine[props.info[i].id]
    if (items) {
      relevantProps.push({
        id: items.product_id,
        results: items.results,
        name: props.info[i].name,
        category: props.info[i].category,
        description: props.info[i].description,
        defaultPrice: props.info[i].default_price
      })
    }
  }

  return (
    <div className="productCard">
      {relevantProps.map((each, index) =>
        <EachCard info={each} key={index} />)}
    </div>
  )
}

export default ProductCard;