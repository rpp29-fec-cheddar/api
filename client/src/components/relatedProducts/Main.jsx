import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';
import ProductCard from './ProductCard.jsx';
import YourOutfit from './YourOutfit.jsx';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      productId: '28212',
      productInfo: [],
      productDetailInfo: []
    }
  }

  //props will be the Product ID coming from Overview

  componentDidMount() {
    $.ajax({
      url: 'http://localhost:4000/relatedProducts/relatedProducts',
      type: 'GET',
      data: {id: this.state.productId},
      success: (data) => {
        console.log('Success', data);
        this.setState({
          productInfo: data
        })
      },
      error: (data) => {
        console.log('Error', data);
      }
    });

    $.ajax({
      url: 'http://localhost:4000/relatedProducts/styles',
      type: 'GET',
      data: {id: this.state.productId},
      success: (data) => {
        console.log('SECOND DATA', data);
        this.setState({
          productDetailInfo: data
        })
      },
      error: (data) => {
        console.log('SECOND ERROR', data);
      }
    });
  }

  render() {
    return(
      <div>
        <h2>Related Products</h2>
        {console.log('Product Info', this.state.productInfo)}
        {console.log('Product DEtail Info', this.state.productDetailInfo)}
        <ProductCard info={this.state.productInfo} detailInfo={this.state.productDetailInfo}/>
        <h2 className="yourOutfitTitle">Your Outfit</h2>
        <YourOutfit />
      </div>
    )
  }
}

export default Main;