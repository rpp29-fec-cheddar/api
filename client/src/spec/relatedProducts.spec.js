import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../components/relatedProducts/Main.jsx';
import ProductCard from '../components/relatedProducts/ProductCard.jsx';
import 'regenerator-runtime/runtime';

describe('RelatedProduct', () => {
  test('Test that Main renders', () => {
    const component = renderer.create(<Main ></Main>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});