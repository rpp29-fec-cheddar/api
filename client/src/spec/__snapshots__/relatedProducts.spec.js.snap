// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedProduct Test that Main renders 1`] = `
<div>
  <h2>
    Related Products
  </h2>
  <div
    className="productCard"
  />
  <h2
    className="yourOutfitTitle"
  >
    Your Outfit
  </h2>
  <div
    className="yourOutfitCard"
  >
    <div
      className="img"
    >
      X
    </div>
    <div
      className="cardCategory"
    >
      Category
    </div>
    <div
      className="expProductName"
    >
      Expanded product name with extra text
    </div>
    <div
      className="cardPrice"
    >
      $80.00
    </div>
    <div
      className="cardRating"
    >
      *****
    </div>
  </div>
</div>
`;
